From 1b8b12d8925b330dfaff6e1cc26f01753711de8d Mon Sep 17 00:00:00 2001
From: "hazemkora660@gmail.com" <hazemkora660@gmail.com>
Date: Sat, 4 Jul 2026 15:16:22 +0300
Subject: [PATCH 12/24] chore(payment): temporarily force EGP currency for
 Paymob intention

- Fix error 417 caused by USD cart colliding with EGP integration IDs

- To be reverted once USD integration IDs are created
---
 .../dukkan/payment/data/repository/PaymentRepositoryImpl.kt    | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/payment/src/main/java/com/dukkan/payment/data/repository/PaymentRepositoryImpl.kt b/core/payment/src/main/java/com/dukkan/payment/data/repository/PaymentRepositoryImpl.kt
index e3f5776..9ad88cc 100644
--- a/core/payment/src/main/java/com/dukkan/payment/data/repository/PaymentRepositoryImpl.kt
+++ b/core/payment/src/main/java/com/dukkan/payment/data/repository/PaymentRepositoryImpl.kt
@@ -47,7 +47,8 @@ internal class PaymentRepositoryImpl @Inject constructor(
         val amountInCents = (cartTotal.amount * java.math.BigDecimal("100")).toInt()
         val json = org.json.JSONObject().apply {
             put("amount", amountInCents) 
-            put("currency", cartTotal.currencyCode)
+            // For testing: Force EGP since our integration IDs (5766356, 5766720) are strictly EGP
+            put("currency", "EGP")
             
             // Add the integration IDs here!
             val paymentMethods = org.json.JSONArray().apply {
-- 
2.45.1.windows.1

